pkg/exporter/exporter.go

```diff
@@ -76,6 +76,11 @@ func (b *Exporter) Listen(e <-chan event.Events) {
 
 // handleEvent processes a single Event according to the configured mapping.
 func (b *Exporter) handleEvent(thisEvent event.Event) {
+	if thisEvent.MetricName() == "" {
+		level.Error(b.Logger).Log("msg", "metric name cannot be empty")
```

Comment:
we cannot log at Error level for every event. In many applications you would produce thousands of lines per second, overwhelming the log system and drowning out everything else.

There is already [code that should handle this](https://github.com/prometheus/statsd_exporter/blob/3b062c01c8cde2f06b64968109a62e7f9608f39a/pkg/exporter/exporter.go#L106-L110), can you find out why it is not working in your case?

Reply:
Oh, I think I see it, that is only reached when there is an explicit mapping and _that_ results in an empty metric name.

Could you please consolidate the empty-name handling to be _after_ the mapping? An explicit mapping may still recover from an incoming event without a name, by specifying one explicitly. As for the mapping-present-but-name-empty case, please log at debug level and increment an error counter.

Please add a test or two as well.

Reply:
I did not configure Mapping，so `present` is  false

Reply:
Yes – the exporter should check for the empty name whether `present` or not, _after_ mapping. Checking the incoming event is not right, because when a mapping is there it might "fix" this. I want to preserve this option, since it allows handling some legacy inputs that cannot be handled otherwise.